txpool/txpool_test.go

```diff
@@ -0,0 +1,121 @@
+package txpool_test
+
+import (
+	"testing"
+	"time"
+
+	"go.sia.tech/sunyata"
+	"go.sia.tech/sunyata/chain"
+	"go.sia.tech/sunyata/consensus"
+	"go.sia.tech/sunyata/internal/chainutil"
+	"go.sia.tech/sunyata/internal/walletutil"
+	"go.sia.tech/sunyata/miner"
+	"go.sia.tech/sunyata/txpool"
+	"go.sia.tech/sunyata/wallet"
+)
+
+// TestTPoolFlexibility tests that the txpool can accept transactions that are
+// valid for the previous validation context
+func TestTPoolFlexibility(t *testing.T) {
+	walletStore := walletutil.NewEphemeralStore()
+	wallet := wallet.NewHotWallet(walletStore, wallet.NewSeed())
+	genesisBlock := sunyata.Block{
+		Header: sunyata.BlockHeader{
+			Timestamp: time.Now().Add(-1 * time.Hour),
+		},
+		Transactions: []sunyata.Transaction{
+			{
+				Outputs: []sunyata.Beneficiary{
+					{
+						Address: wallet.NextAddress(),
+						Value:   sunyata.BaseUnitsPerCoin.Mul64(100),
+					},
+				},
+			},
+		},
+	}
+	genesis := consensus.Checkpoint{
+		Block:   genesisBlock,
+		Context: consensus.GenesisUpdate(genesisBlock, sunyata.Work{NumHashes: [32]byte{31: 1}}).Context,
+	}
+	chainStore := chainutil.NewEphemeralStore(genesis)
+	cm := chain.NewManager(chainStore, genesis.Context)
+	tp := txpool.New(genesis.Context)
+
+	if err := cm.AddSubscriber(walletStore, genesis.Context.Index); err != nil {
+		t.Fatalf("failed to subscribe wallet: %s", err)
+	}
+
+	m := miner.New(genesis.Context, wallet.NextAddress(), tp, miner.CPU)
+	if err := cm.AddSubscriber(tp, cm.Tip()); err != nil {
+		t.Fatalf("failed to subscribe txpool: %s", err)
+	}
+	if err := cm.AddSubscriber(m, cm.Tip()); err != nil {
+		t.Fatalf("failed to subscribe miner: %s", err)
+	}
+
+	// Helper function to miner blocks using our miner.
+	mineBlock := func(n int) {
```

Comment:
you should be able to use `chainutil.ChainSim` to handle most of the boilerplate in this test -- refer to `wallet_test.go`